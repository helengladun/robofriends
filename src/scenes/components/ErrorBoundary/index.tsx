import React, {Component, ReactNode} from 'react';

interface IProps {
  children: ReactNode
}

class ErrorBoundary extends Component<IProps> {
  state = {
    hasError: false
  };

  componentDidCatch(): void {
    this.setState(() => ({
      hasError: true
    }));
  }

  render() {
    if (this.state.hasError) {
      return <h1>Ooops. That is not good</h1>;
    }

    return this.props.children;
  }
}

export default ErrorBoundary;