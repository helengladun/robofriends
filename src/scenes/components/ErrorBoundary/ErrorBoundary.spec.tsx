import React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundary from './index';

describe('ErrorBoundary component', () => {
  it('render ErrorBoundary component correctly', () => {
    expect(shallow(<ErrorBoundary />)).toMatchSnapshot()
  });
});
