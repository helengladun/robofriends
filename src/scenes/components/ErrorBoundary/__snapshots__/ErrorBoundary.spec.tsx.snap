// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component render ErrorBoundary component correctly 1`] = `""`;
